### PR TITLE
docs: Renaming RestClientParams to ClientParams and updating docs examples

### PR DESCRIPTION
**Pull request by rakatyal**
[Internal Reference](https://issues.corp.twilio.com/browse/DI-1677)

### CODE REVIEW

File: twilio_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 func TestClient_WithParams(t *testing.T) {
-	client := NewRestClientWithParams(RestClientParams{
+	client := NewRestClientWithParams(ClientParams{
```
Comment (beebzz) [inline twilio_test.go:10]: Would it be worthwhile to add tests for the use of subaccounts?

File: README.md [hunk 2]
```diff
@@ -224,7 +227,8 @@ import (
 )
 
 func main() {
-    serviceSid := "ZSxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+	// serviceSid should be in format "ZSxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
```
Comment (beebzz) [inline README.md:230]: nit: indent alignment

File: README.md [hunk 3]
```diff
@@ -44,47 +44,56 @@ Getting started with the Twilio API couldn't be easier. Create a
...
-    accountSid := "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
-    authToken := "YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY"
-    client := twilio.NewRestClientWithParams(twilio.RestClientParams{
-    	Username: accountSid,
-    	Password: authToken,
-    })
+	// This will look for `TWILIO_ACCOUNT_SID` and `TWILIO_AUTH_TOKEN` variables inside the current environment to initialize the constructor
+	// You can find your Account SID and Auth Token at twilio.com/console
+	// `TWILIO_ACCOUNT_SID` should be in format "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+    client := twilio.NewRestClient()
```
Comment (childish-sambino) [inline README.md:58]: nit: mixing spaces and tabs. Should update all examples in this file to just use tabs.

File: README.md [hunk 4]
```diff
@@ -44,47 +44,56 @@ Getting started with the Twilio API couldn't be easier. Create a
...
+    authToken := "YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY"
+    client := twilio.NewRestClientWithParams(twilio.ClientParams{
+    	Username: accountSid,
+    	Password: authToken,
+    })
 }
 ```
-Make API calls with a SubAccount by setting the `AccountSid` field in the `twilio.RestClientParams`:
+
+### Using SubAccount
```
Comment (childish-sambino) [inline README.md:79]: nit: should be `Subaccount` (lowercase 'a')

File: twilio_test.go (hunk 1 above)
Comment (rakatyal) [reply twilio_test.go:10]: I thought this test was testing the use of subAccountSid vs the other one is testing the main account?

File: README.md (hunk 2 above)
Comment (rakatyal) [reply README.md:230]: Fixed. I updated all the examples to use "tabs" instead of "spaces".

File: README.md (hunk 3 above)
Comment (rakatyal) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (rakatyal) [reply README.md]: Fixed.

File: README.md [hunk 5]
```diff
@@ -44,47 +44,56 @@ Getting started with the Twilio API couldn't be easier. Create a
...
+
+Subaccounts in Twilio are just accounts that are "owned" by your account. Twilio users can create subaccounts to help separate Twilio account usage into different buckets.
+
+If you wish to make API calls with a SubAccount, you can do so by setting the `AccountSid` field in the `twilio.ClientParams`:
```
[the remark this answers is not in the document]
Comment (rakatyal) [reply README.md]: Fixed.

File: README.md [hunk 6]
```diff
@@ -516,18 +520,18 @@ apiSecret := os.Getenv("TWILIO_API_SECRET")
...
 
 jwtToken := jwt.CreateAccessToken(params)
 voiceGrant := &jwt.VoiceGrant{
-    Incoming: jwt.Incoming{Allow: true},
-    Outgoing: jwt.Outgoing{
-        ApplicationSid:    applicationSid,
-    },
+	Incoming: jwt.Incoming{Allow: true},
+	Outgoing: jwt.Outgoing{
+    	ApplicationSid:    applicationSid,
```
Comment (childish-sambino) [inline README.md:533]: nit: still some spaces scattered throughout.

**Review by childish-sambino — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (rakatyal) [reply README.md]: Hmm somehow my find and replace missed some. Should be resolved now.

File: twilio_test.go (hunk 1 above)
Comment (beebzz) [reply twilio_test.go:10]: Ah alright, disregard then

**Pull request merged**